### PR TITLE
Add Linear Probing experiment pipeline for BirdSet datasets

### PR DESCRIPTION
**Pull request by The-Mats**
This PR adds multilabel linear probing experiments on the BirdSet datasets.
First all training, valid and eval samples are converted into embeddings using the `birdset_embedding_datamodule`.
Then a linear layer is trained using `MultilabelModule`.
Evaluation is done with the soundscape split of the BirdSet datsets.

This approach does not allow argumentations due to the precomputing of the embeddings.

## Results

| Model | cmAp |
|-------| -------|
| BEATs| 0.08|
| convnext | 0.04 |
| perch | 0.16 |

## Prior stuff 
To use BirdSet data we need to make sure to do the correct transformation of the audio so that multilabel can work correctly. 
Using decode=True when using _load_data() does load the audio array but causes errors in _smart_sampling()  (df = pd.DataFrame(dataset)) so I would suggest converting it later like usually done in the BS pipeline. 

Running the entire transformations is also not ideal as it doesn't fit into the structure of compute_and_update_embedding() in the embedding_datamodule.py. Additionally, spectrogram creation is not needed as it happens in the models themselves when using get_embeddings().

Therefore, I would suggest only integrating feature_extractor and decoder as I believe they are needed for multilabel.

- [x] Fix errors when initializing HSN dataset
- [x] Integrate decoder if needed
- [x] Move decoder to birdset_embedding_datamodule
- [x] Integrate feature_extractor if needed: Not needed
- [x] Fix remaining errors when running the BEATS-HSN experiment
- [x] Check for BEANS compatibility (Fix some problems like naming of sampling_rate)
- [x] Test with other models 
- [x] Setup for other BS test sets
- [ ] <s> Add Waveform augmentations
- [ ] See if Spectrogram augmentations can be added
- [ ] Optimize hyperparams for ConvNeXt to see how it compares to BS
- [ ] Optimize hyperparams for BEATS
- [ ] Do a full run for the paper </s>

There is an error when using multiple workers for the dataset that I couldn't solve yet (n_workers=1) for now!

~~At the moment there is a freezing error after smart sampling which does not occur when training the beans_HSN.yaml in local/HSN~~

### CODE REVIEW

**Comment by The-Mats** [comment]
@raphaelschwinger The DT experiments from birdset_neurips24 seem to have different params (batch size) per model and dataset. What batch size and so on should we use? The same for all?

**Comment by raphaelschwinger** [comment]
@XgamerTV The batch size is depending on the model size. Lets stick with one for all to keep it simple for now. 32 should fit on our GPUs.

**Comment by The-Mats** [comment]
The results of BEATS on HSN is a lot lower than the ones in the BirdSet paper (12% T1) but since only the classifier is trained this may be expected 🤔
The perch performance is a lot better with 37% ([WandB Report](https://wandb.ai/deepbirddetect/BioFoundation/runs/lzxofbpn?nw=nwusermatsaustralia))
I also started testing with NIPS4BPlus

**Comment by The-Mats** [comment]
What is left to do on this PR? Right now there is a beats experiment for all BirdSet datasets but other models can be easily added.

**Comment by raphaelschwinger** [comment]
@XgamerTV,
Play around with the hyperparameters a bit (learning rate, batch_size, number of layers of the classifier) and then run the experiments for all models and all datasets on the fixed set of HP.

**Comment by The-Mats** [comment]
@raphaelschwinger Ok I would start with one seed for now as it will take a lot of time 👍🏼
